### PR TITLE
Forms cleanup

### PR DESCRIPTION
**Pull request by sandhose**
This does almost everything layed out in https://github.com/vector-im/compound/issues/249

It also includes #104 and #106 

Main exception is that I did not touch the Radio, Toggle and Search components.

This is best reviewed commit by commit! Happy to split it up in smaller PRs if that's preferred.

### CODE REVIEW

File: src/components/Form/Controls/MFA/MFA.module.css [hunk 1]
```diff
@@ -14,26 +15,98 @@ See the License for the specific language governing permissions and
...
-  opacity: 0;
+.control {
+  all: unset;
+
+  /* TODO: a font stack for monospace fonts should be defined in the design tokens
+   * Font stack from https://qwtel.com/posts/software/the-monospaced-system-ui-css-font-stack/#the-monospaced-system-ui-css-font-stack */
+  font-family: ui-monospace, Menlo, Monaco, "Cascadia Mono", "Segoe UI Mono",
+    "Roboto Mono", "Oxygen Mono", "Ubuntu Monospace", "Source Code Pro",
+    "Fira Mono", "Droid Sans Mono", "Courier New", monospace;
+  font-weight: bold;
```
Comment (germain-gg) [inline src/components/Form/Controls/MFA/MFA.module.css:41]: I think we should follow what Element Web currently does. => using `Inconsolata` and the same font stakc as them

File: src/components/Form/Controls/MFA/MFA.module.css [hunk 2]
```diff
@@ -14,26 +15,98 @@ See the License for the specific language governing permissions and
...
+  border-radius: 0.5rem;
+  padding: var(--cpd-space-3x) var(--cpd-space-4x);
+}
+
+@media (hover) {
+  .control:hover ~ .digit {
+    border-color: var(--cpd-color-border-interactive-hovered);
+
+    /** TODO: have the shadow in the design tokens */
+    box-shadow: 0 1.2px 2.4px 0 rgb(0 0 0 / 15%);
```
Comment (germain-gg) [inline src/components/Form/Controls/MFA/MFA.module.css:75]: Is the shadow always suppose to be black as is even in dark mode?
@janogarcia ?

**Review by germain-gg — COMMENTED** [review]
Did a first pass review, overall looks good!
Added a couple of questions, will do another one commit by commit later

File: src/components/Form/Controls/MFA/MFA.module.css (hunk 1 above)
Comment (sandhose) [reply src/components/Form/Controls/MFA/MFA.module.css:41]: Should I just copy the font stack or wait for it to be formalised in the design tokens?

**Comment by sandhose** [comment]
I rebased on top of main to add the `autodocs` to all stories like in #107 

The 3 added commits are for:

 - the `<Radio />` input, cleaning it up and making it like all other controls
 - the `<Checkbox />` input, cleaning it up and making it like all other controls
 - a new `<InlineField />` component for use with controls which are on the same line as their label (checkbox, radio, toggle)

File: src/components/Form/Controls/MFA/MFA.module.css (hunk 1 above)
Comment (germain-gg) [reply src/components/Form/Controls/MFA/MFA.module.css]: I think it deems to be part of the design tokens in the long run

**Review by germain-gg — APPROVED** [review]

**Comment by germain-gg** [comment]
Thank you, I think that looks like a good improvement, and fixes a lot of the initial inconsistencies of form authoring.

I'm meeting up with @kerryarchibald tomorrow, so we might discuss that a bit further

**Comment by germain-gg** [comment]
Do you consider this would fix https://github.com/vector-im/compound-web/pull/108 ?

**Comment by sandhose** [comment]
> Do you consider this would fix [vector-im/compound#249](https://github.com/vector-im/compound/issues/249) ?

Yes, with the exception of the Search component, which I did not touch

**Comment by sandhose** [comment]
I implemented the readonly state for Toggle, Checkbox, Radio components as per https://github.com/vector-im/compound/issues/251

File: src/index.ts [hunk 3]
```diff
@@ -40,18 +36,45 @@ export {
...
-export { Submit } from "./components/Form/Submit";
 export { Search } from "./components/Search/Search";
-export { Toggle } from "./components/Toggle/Toggle";
 export { Tooltip } from "./components/Tooltip/Tooltip";
-export { ValidityState } from "./components/Form/ValidityState";
+
+export {
+  TextControl,
+  /* Legacy export name */
+  TextControl as Control,
```
Comment (sandhose) [inline src/index.ts:47]: I checked with EC and ESS and no one is using the form right now, other than MAS. @germain-gg: should I get rid of the "legacy" export names? (`Checkbox`, `Radio` and `Toggle` below as well)

**Pull request merged**